### PR TITLE
Fixed numerous AutoSklearn issues

### LINKED ISSUE #77: AutoSKLearn or top-k-count bug
AutoSKLearn should get a perfect score in top-k-count for training data. It is not. Either AutoSKLearn is wrong or the metric is.

### PR DESCRIPTION
**Pull request by orionw**
- Fixed sorting, 0 distance metric case, and k options.

Note that we the top-k count on the small dataset comes out to be exactly 1.0.  I attribute that to the small amount of data - the odds that it would get the exact 2 right out of 140 is small and there are only ~6 test sets.  Running it on the large dataset gives me X (I'll update with the real number when it finishes)

Closes #77, #79

### CODE REVIEW

**Comment by bjschoenfeld** [comment]
Waiting on #86
